lib/active_model/serializer.rb

```diff
@@ -131,6 +320,36 @@ def success?
       true
     end
 
+    # Return the +attributes+ of +object+ as presented
+    # by the serializer.
+    def attributes(requested_attrs = nil, reload = false)
+      @attributes = nil if reload
+      @attributes ||= self.class._attributes_data.each_with_object({}) do |(key, attr), hash|
+        next if attr.excluded?(self)
+        next unless requested_attrs.nil? || requested_attrs.include?(key)
+        hash[key] = attr.value(self)
+      end
+    end
+
+    # @param [JSONAPI::IncludeDirective] include_directive (defaults to the
+    #   +default_include_directive+ config value when not provided)
+    # @return [Enumerator<Association>]
+    #
+    def associations(include_directive = ActiveModelSerializers.default_include_directive, include_slice = nil)
+      include_slice ||= include_directive
+      return unless object
+
+      Enumerator.new do |y|
+        self.class._reflections.values.each do |reflection|
+          next if reflection.excluded?(self)
+          key = reflection.options.fetch(:key, reflection.name)
```

Comment:
NOTE:  `key` should always be set since

```ruby
    def self.associate(reflection)
      key = reflection.options[:key] || reflection.name
      self._reflections[key] = reflection
    end
```

which mean we could safely change `.values.each do |reflection|` to `.each do |key, reflection|`